tests/test_edgeql_expressions.py

```diff
@@ -4778,3 +4778,59 @@ async def test_edgeql_normalization_missmatch_01(self):
                 edgedb.EdgeQLSyntaxError, "Unexpected type expression"):
 
             await self.con.query('SELECT <tuple<"">>1;')
+
+    async def test_edgeql_assert_single_01(self):
+        await self.con.execute("""
+            INSERT User {
+                name := "He Who Remains"
+            }
+        """)
+
+        await self.assert_query_result("""
+            SELECT assert_single((
+                SELECT User { name } FILTER .name ILIKE "He Who%"
+            ))
+        """, [{
+            "name": "He Who Remains",
+        }])
```

Comment:
Could you add a test with a computed property, and a test that does some computation on a shape result (instead of directly outputting it)?

Reply:
> and a test that does some computation on a shape result (instead of directly outputting it)?

+1